### PR TITLE
Load the correct config.json and genesisblock.json based on the NODE_ENV

### PR DESCRIPTION
**Pull request by reasonthearchitect**
Currently, users must cp files from the test package in order to run the test suite. Additionally, before starting the server users must dropdb and createdb before running node app.js.

In this pull request I have created a file, on the root, called alias.js. This file should be called before all others. This file will check the NODE_ENV.toLowercase() var to see if it is of type 'test'. IF it is the file /test/config.json and /test/genesisBlock.json rather than the root directory files. If it is any other type (or not set) it will load up the root files. Once loaded, the files are cached as singletons.

Additionally, I have updated and call to require('<path>/[config || genesisBlock}.json in all files to ensure that they receive the right version of configuration for the environment.

Also, i have added better-npm-run to the project in order to write cleaner npm run commands. This is not required, however, it seems that the file becomes more legible.

Finally. the .eslintignore file I have added alias as a global. This simplifies its use, however, it can easily be created as a module export if that is more inline with your teams opinion. .gitignore has added package-lock.json.

Hope to here from you soon.
Rhys

### CODE REVIEW

File: test/api/transactions.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
-var node = require('../node.js');
-var http = require('../common/httpCommunication.js');
-var ws = require('../common/wsCommunication');
-var sendLiskTrs = require('../common/complexTransactions.js').sendLISK;
-var transactionSortFields = require('../../sql/transactions').sortFields;
-var modulesLoader = require('../common/initModule').modulesLoader;
-var transactionTypes = require('../../helpers/transactionTypes.js');
-var genesisblock = require('../genesisBlock.json');
+var node 										= alias.require('test/node.js');
```
Comment (MaciejBaj) [inline test/api/transactions.js]: prefer the regular indents

File: test/api/transport/transport.blocks.js [hunk 2]
```diff
@@ -1,11 +1,10 @@
 'use strict';
 
-var node = require('../../node.js');
-var ws = require('../../common/wsCommunication');
-
-var genesisblock = require('../../genesisBlock.json');
-var verify = require('../../../modules/blocks/verify.js');
-var bson = require('../../../helpers/bson.js');
+var node 					= alias.require('test/node.js');
```
Comment (MaciejBaj) [inline test/api/transport/transport.blocks.js]: prefer the regular indents

File: test/common/initModule.js [hunk 3]
```diff
@@ -1,26 +1,22 @@
...
-var Logger = require(dirname + '/logger.js');
-var z_schema = require('../../helpers/z_schema.js');
-var cacheHelper = require('../../helpers/cache.js');
-var Cache = require('../../modules/cache.js');
-var ed = require('../../helpers/ed');
-var jobsQueue = require('../../helpers/jobsQueue');
-var Transaction = require('../../logic/transaction.js');
-var Account = require('../../logic/account.js');
-var Sequence = require('../../helpers/sequence.js');
+var Sequence 			= alias.require('helpers/sequence.js');
```
Comment (MaciejBaj) [inline test/common/initModule.js]: prefer the regular indents

File: test/common/initModule.js [hunk 4]
```diff
@@ -182,22 +178,22 @@ var modulesLoader = new function () {
...
-			{crypto: require('../../modules/crypto')},
-			{delegates: require('../../modules/delegates')},
-			{loader: require('../../modules/loader')},
-			{multisignatures: require('../../modules/multisignatures')},
-			{peers: require('../../modules/peers')},
-			{signatures: require('../../modules/signatures')},
-			{system: require('../../modules/system')},
-			{transactions: require('../../modules/transactions')},
-			{transport: require('../../modules/transport')}
+			{accounts: 					alias.require('modules/accounts')},
```
Comment (MaciejBaj) [inline test/common/initModule.js]: prefer the regular indents

File: test/api/transactions.js (hunk 1 above)
Comment (reasonthearchitect) [reply test/api/transactions.js]: I will make sure to follow that in the future.

File: test/common/initModule.js (hunk 4 above)
Comment (reasonthearchitect) [reply test/common/initModule.js]: I have reverted the files to use standard indentation and pushed it to the repo.

**Comment by reasonthearchitect** [comment]
I have reverted the tabing on require() to spaces as per requested. Additionally, while writing unit tests for helpers/config.js last night, I noted that you can load another config file from the command line. As I was not aware if this is being done somewhere (production i suppose) I have done the command line check in the alias.js file. This will ensure that we are able to still override the configuration file. However, I did not see and option for genesisBlock.json overriding.

As a result, I have updated this branch with the correct logic to still ensure loading of a configuration file provided in the command line args.

**Comment by reasonthearchitect** [comment]
Guys.... any thoughts on this one... are you still interested in it?

**Comment by SargeKhan** [comment]
Hey @reasonthearchitect, I have reviewed the pull request and discussed it with @willclarktech, and I think that we shouldn't follow `alias.rewire` the approach of importing files. The rationale behind it is that this solution adds unnecessary obscurity in our code, because we ask `alias.rewire` to implicitly decide something for us, as opposed to us being more explicit in our code.

I really like the idea of using different configuration and genesis block based on ENV variable, and I would be glad to merge it but not the `alias.require` part of this PR. Let me know if you have any questions. 

Thanks for your time and great effort!

**Comment by SargeKhan** [comment]
As discussed on gitter, I will close this PR and @reasonthearchitect will open a new PR which incorporates the changes which impact only config.json and genesisBlock.json

**Pull request closed without merging**